### PR TITLE
SMS View

### PR DESCRIPTION
**Pull request by bclymer**
https://github.com/jariz/Noti/issues/23

Let's just say I'm not a design wizard, however, I got most of the backend working. It will automatically prefer you're primary device that supports SMS.

How to launch:
screenshot 2016-11-25 20 57 42

List of threads:
screenshot 2016-11-25 20 57 21

Conversation view:
screenshot 2016-11-25 20 57 32

I ended up being a fairly significant amount of refactoring. I've been working in Swift for a number of years professionally (although never for macOS, just iOS) but I hope you don't mind that I cleaned up and organized some stuff. I also updated to Swift 3 (to support Xcode 8).

### CODE REVIEW

**Comment by jariz** [comment]
Oh wow I already had swift 3 working in the 0.2 branch 😥. 
This looks great though. I'll take a better look later.

**Comment by jariz** [comment]
Looking closer into this, I'd really like you to try to merge `0.2` into your branch, as it's going to be merged with master very soon (before this gets merged, at least), and I can't really start looking into merging this with master otherwise.
Sorry 😞 .

But, I'll ask @mirshko to do a design for this!

**Comment by jariz** [comment]
Also, don't mind the clean up at all, I'm not a professional swift dev by any means, so that's very much appreciated.

**Comment by mirshko** [comment]
@bclymer @jariz I'm on it! Ill get started tonight!

**Comment by bclymer** [comment]
Because of how many files I moved around, doing an actual `git merge` isn't going to work (more accurately it will be too much work). My recommended path would be to wait until your `0.2` hits `master`, and then I would manually apply your commits from `0.2` into my branch.

The other solution would be me checking out `master` once `0.2` is merged in, and manually apply all of my commits (ignoring the Swift 3 migration).

The only reason I recommend the former is number / size of commits. But that does have the downside of messing up your git history, as we'd essentially just accept my branch as the new master (still in your repo), but your actual git commits in `0.2` would be lost, they would just happen to exist from the manual port.

TL;DR - Once you merge `0.2` I will find a way to make my branch mergable. Right now doing a `git merge` is too much work given how much the config files conflict.

**Comment by mirshko** [comment]
Here is V1 of the conversation view! Message view coming tomorrow!

noti conversation view

**Comment by mirshko** [comment]
Sorry for the delay! Here are both conversation and messages views! 

noti conversation view
noti messages view

**Comment by mirshko** [comment]
@jariz Let me know what you think!

**Comment by bclymer** [comment]
@mirshko those look great. I haven't forgotten about this. I'm currently waiting on 2.0 to get into `master` to be my merge, then I'll get going on these changes.

**Comment by mirshko** [comment]
@bclymer Thanks man :) Lemme attach the Sketch files here to help with the spec.

**Comment by mirshko** [comment]
[Noti SMS Panels.zip](https://github.com/jariz/Noti/files/649971/Noti.SMS.Panels.zip)

Here is the Sketch file!

**Comment by jariz** [comment]
Going to try to get 0.2 ready this week, sorry for delaying this !

**Comment by jariz** [comment]
Hey, so, 0.2 is ready! Hope the merge conflict isn't too hard to resolve 😛
